src/KurrentDB.Client/Streams/KurrentDBClient.Append.cs

```diff
@@ -195,7 +187,7 @@ IWriteResult HandleWrongExpectedRevision(
 				actualStreamRevision
 			);
 
-			if (operationOptions.ThrowOnAppendFailure) {
+			if (operationOptions.ThrowOnAppendFailure == true) {
```

Comment:
Minor, but had me go "why?". Did it become nullable perhaps?

Reply:
Yes, and no 🙂 

Yes because `ThrowOnAppendFailure` indeed is nullable now, as it can be provided or not by the user. I prefer to keep it nullable to have clear information whether the user sets it or not. 

No because it's different OperationSettings than before. Previously we used here `KurrentDBClientOperationOptions`, which are general settings for the client, and it contains various options that may not be related to specific operations. Previously they were passed even to read operations, where `ThrowOnAppendFailure` wouldn't make any sense. I replaced them with more explicit and contextual `AppendToStreamOptions` and generic `OperationOptions`.